src/Core/Scheduler/Providers/GpxUtils.js

```diff
@@ -16,78 +16,153 @@ function _gGpxToWTrackPointsArray(gpxXML) {
+    color: 0xff0000,
+    emissive: 0xff0000,
+    roughness: 1.0,
+    metalness: 0.0,
+    displacementMap: dataTexture,
+});
+
+function updatePathRadius(renderer, scene, camera) {
+    const distance = getDistance(this, camera);
+    this.material.displacementScale = distance / renderer.getSize().height;
+    this.material.depthTest = distance < this.geometry.boundingSphere.radius * 2;
+    this.material.needsUpdate = true;
+}
+
+function _gpxToWTrackPointsMesh(gpxXML, crs) {
     var trackPts = _gGpxToWTrackPointsArray(gpxXML);
 
     if (trackPts.length) {
-        const positions = new Float32Array(trackPts.length * 3);
-        for (var i = 0; i < trackPts.length; i++) {
-            const pos = _gpxPtToCartesian(trackPts[i]);
-            pos.toArray(positions, 3 * i);
+        const points = [];
+
+        if (gpxXML.center === undefined) {
+            gpxXML.center = _gpxPtToCartesian(trackPts[0], crs);
+        }
+
+        for (const trackPt of trackPts) {
+            points.push(_gpxPtToCartesian(trackPt, crs).sub(gpxXML.center));
         }
 
-        const line = new THREE.Line();
-        line.geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
-        line.material.color.setRGB(1, 0, 0);
-        line.material.linewidth = 100;
+        // Due to limitations in the ANGLE layer, with the WebGL renderer on Windows platforms linewidth will always be 1 regardless of the set value.
+        // For this reason, we use TubeBufferGeometry to change radius with displacement map.
+        const path = new PathCurve(points);
+        const geometryPath = new THREE.TubeBufferGeometry(path, points.length - 1, 0.25, 4, false);
+        const pathMesh = new THREE.Mesh(geometryPath, materialLine);
 
-        return line;
+        // Scale radius line in function of distance, Use displacement map to scale radius
+        pathMesh.onBeforeRender = updatePathRadius;
+        return pathMesh;
     } else {
         return null;
     }
 }
 
-function _gpxToMesh(gpxXML) {
+function _gpxToMesh(gpxXML, crs) {
     if (!gpxXML) {
         return undefined;
     }
 
     var gpxMesh = new THREE.Object3D();
 
     // Getting the track points
-    var trackPts = _gpxToWTrackPointsMesh(gpxXML);
+    var trackPts = _gpxToWTrackPointsMesh(gpxXML, crs);
 
     if (trackPts) {
         gpxMesh.add(trackPts);
     }
 
     // Getting the waypoint points
-    var wayPts = _gpxToWayPointsMesh(gpxXML);
+    var wayPts = _gpxToWayPointsMesh(gpxXML, crs);
 
     if (wayPts) {
         gpxMesh.add(wayPts);
     }
 
+    gpxMesh.position.copy(gpxXML.center);
+    gpxMesh.updateMatrixWorld();
+    // gpxMesh is static data, it doens't need matrix update
+    gpxMesh.matrixAutoUpdate = false;
+
     return gpxMesh;
 }
 
-export default function loadGpx(urlFile, networkOptions) {
-    return Fetcher.xml(urlFile, networkOptions).then(gpxXML => _gpxToMesh(gpxXML));
+export default function loadGpx(urlFile, crs, networkOptions) {
```

Comment:
Don't you think you should include `networkOptions` in a `options` object that would also allow to configure a few things:
- use 'pin' or not
- line size (because with your change the user cannot simply set : `line.material.linewidth`)